### PR TITLE
Updating our "is probably over" heuristic with an idea from @kinch2002.

### PR DESCRIPTION
**Pull request by jfly**

### CODE REVIEW

File: WcaOnRails/app/models/competition.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ class Competition < ApplicationRecord
            allow_nil: true,
            with_model_currency: :currency_code
 
-  scope :not_over, -> { where("end_date >= ?", Date.today) }
+  scope :not_over, -> { where("end_date >= ? AND results_posted_at IS NULL", Date.today) }
```
Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: Gah, I *hate* keeping these scopes in sync with our boolean model methods. @larspetrus, any ideas to help us not repeat ourselves here?

Comment (larspetrus) [inline WcaOnRails/app/models/competition.rb]: You could probably rewrite `is_probably_over?` to use the `not_over` scope. The price is an extra DB query.

Are they really synced when `end_date.nil?`?

Comment (jfly) [inline WcaOnRails/app/models/competition.rb]: > The price is an extra DB query.

That could get expensive if we're doing it in a loop over a bunch of competitions =(

> Are they really synced when end_date.nil??

No, but I don't really have any opinion on what should happen when `end_date` is `NULL`. Making `not_over` be the opposite of `is_probably_over` seems like a reasonable behavior.

Comment (larspetrus) [inline WcaOnRails/app/models/competition.rb]: OK. The I'm out of ideas. You can see the separate methods as a speed optimization.

**Comment by jfly** [comment]
I don't think fixing this transient issue is worth the added complexity, so I'm going to close this PR without merging it.

**Pull request closed without merging**